### ISSUE TITLE
CCColor color components can not be modified.

### ISSUE THREAD

**Issue by Birkemose** [cocos2d/cocos2d-objc#1029]
The color components are read only. Any reason for that?

**Comment by haligalaph** [comment]
With readwrite rgb properties you will be able to change color of CCNode without calling setColor: method, result in node's inconsistent state. In this case It is better to add CCMutableColor class.

**Comment by slembcke** [comment]
Right. I vaguely remember this being discussed when we first made the CCColor class. Making it mutable would mean copy properties everywhere and a separate CCMutableColor class. I thought I remembered everybody thinking that going the extra step for the color class would be a little silly if all the properties returned immutable objects anyway.
